be/btree.c

```diff
@@ -2138,37 +2327,53 @@ M0_INTERNAL void m0_be_btree_save_inplace(struct m0_be_btree        *tree,
 					  struct m0_be_tx           *tx,
 					  struct m0_be_op           *op,
 					  const struct m0_buf       *key,
+					  uint64_t                   ver,
 					  struct m0_be_btree_anchor *anchor,
 					  bool                       overwrite,
 					  uint64_t                   zonemask)
 {
+	struct btree_pair_ver bpv = BPV_NONE;
+
 	M0_ENTRY("tree=%p zonemask=%"PRIx64, tree, zonemask);
 	M0_PRE(tree->bb_root != NULL && tree->bb_ops != NULL);
 	M0_PRE(key->b_nob == be_btree_ksize(tree, key->b_addr));
 
-	btree_save(tree, tx, op, key, NULL, anchor, overwrite ?
+	if (ver != 0) {
+		bpv_tbs_clear(&bpv);
+		bpv_ver_set(&bpv, ver);
+		M0_ASSERT(bpv_invariant(&bpv));
```

Comment:
why do we need to call this invariant unless we obtained the metadata from persistent storage?
bpv lives inside the stack memory.